system/exec_cmd_runner_windows.go

```diff
@@ -1,14 +1,18 @@
 package system
 
 import (
+	"fmt"
 	"os/exec"
 	"sort"
 	"strings"
 )
 
-func newExecCmd(name string, args ...string) *exec.Cmd {
-	args = append([]string{name}, args...)
-	return exec.Command(`powershell`, args...)
+func newExecCmd(isScript bool, name string, args ...string) *exec.Cmd {
+	if isScript {
+		return exec.Command("powershell", append([]string{fmt.Sprintf("%s.ps1", name)}, args...)...)
```

Comment:
This looks broken, PowerShell will **only** execute files with the `.ps1` extension and if `name` is the path to a `ps1` file then you may want to add the `-File` flag (e.g. `PowerShell.exe -File NAME [ARGS...]`).
 
Also, blindly adding the `.ps1` es will break scripts that already have a `.ps1` extension what you would want is:
 
```golang
// File exts are case-insensitive on Windows
if strings.ToLower(filepath.Ext(name)) != ".ps1" {
	name += ".ps1"
}
```

Reply:
Ideally, it would be neat if we did not append the ".ps1" at the end of the script in the first place and expect the operator to provide the file-extension.

Reply:
@ashwin-venkatesh If I remember correctly that didn't play well with BOSH releases.

Reply:
There is an ugly hack in the BOSH-Agent that gets around this by essentially eval'ing the file, but it's painful for operators as any error is reported with something like: `Error at Line 1 Char 1` and provides no context.

https://github.com/cloudfoundry/bosh-agent/blob/master/agent/compiler/compiler_windows.go#L10

Reply:
@charlievieth Oh. That makes sense 👍

Reply:
One solution would be to copy the script to a temp file with the `.ps1` extension and run in the same directory as the script (maybe `foo-script` => `foo-script.ps1` or `tmp.foo-script.ps1`), but you probably want to keep the copy in the same directory as the source because of stuff like: `$scriptPath = split-path -parent $MyInvocation.MyCommand.Definition`.

Reply:
It sounds like code that makes sense while being written but like hieroglyphics about 5 minutes later. I mean I really like the solution but at the same point of time, it has that soupçon on un-maintainability. You know what I mean?